src/main/java/pl/put/bulidinginfo/models/RoomCompound.java

```diff
@@ -23,10 +29,35 @@ public List<Location> getLocations() {
         return locations;
     }
 
+    public Type getType() { return this.type; }
+
     @Override
     public Float getArea() {
         return this.getLocations().stream().map(location -> location.getArea()).reduce(0f, Float::sum);
     }
-    
+
+    @Override
+    public Float getHeatingPerCubeMeter() {
+        throw new UnsupportedOperationException("This method cannot be used on BUNCH OF ROOMS objects. This method is only used on the SINGLE ROOM object.");
+    }
+
+    @Override
+    public List<Location> getLocationsWithHigherHeatingRate(Float thresholdRate) {
+        if (getType() != Type.BUILDING) {
+            throw new UnsupportedOperationException("This method cannot be used on this object. This method is only used on the BUILDING object.");
+        }
+
+        List<Location> filteredLocations = new ArrayList<>();
+        for (Location floor : getLocations()) {
+            if (floor instanceof RoomCompound) {
+                for (Location room : ((RoomCompound) floor).getLocations()) {
+                    if (room.getHeatingPerCubeMeter() > thresholdRate) {
```

Comment:
Lepiej ten warunek przenieść do Room jeśli jest spełniony to zwracać listę jedno elementową z pokojem, jeśli nie to pustą, a tutaj z kolei tylko agregować listy